libraries/AP_HAL_ChibiOS/hwdef/HitecMosaic/hwdef.dat

```diff
@@ -0,0 +1,172 @@
+g# hw definition file for processing by chibios_pins.py
+
+# MCU class and specific type
+MCU STM32F303 STM32F303xC
+
+# bootloader starts firmware at 26k
+FLASH_RESERVE_START_KB 26
+
+# store parameters in pages 11 and 12
+define STORAGE_FLASH_PAGE 11
+define HAL_STORAGE_SIZE 800
+
+# board ID for firmware load
+APJ_BOARD_ID 1016
+
+# setup build for a peripheral firmware
+env AP_PERIPH 1
+
+# enable watchdog
+define HAL_WATCHDOG_ENABLED_DEFAULT true
+
+# crystal frequency
+OSCILLATOR_HZ 24000000
+
+define CH_CFG_ST_FREQUENCY 100000
+define CH_CFG_ST_TIMEDELTA 0
+
+# assume the 256k flash part for now
+FLASH_SIZE_KB 256
+
+# order of UARTs
+SERIAL_ORDER USART2 EMPTY EMPTY USART3
+
+define HAL_CAN_POOL_SIZE 6000
+
+STDOUT_SERIAL SD2
+STDOUT_BAUDRATE 57600
+
+# USART2, debug
+PA2 USART2_TX USART2 SPEED_HIGH NODMA
+PA3 USART2_RX USART2 SPEED_HIGH NODMA
+
+# USART3 for GPS
+PB10 USART3_TX USART3 SPEED_HIGH NODMA
+PB11 USART3_RX USART3 SPEED_HIGH NODMA
+
+# only one I2C bus in normal config, no external pullup
+PA15 I2C1_SCL I2C1 PULLUP
+PB7  I2C1_SDA I2C1 PULLUP
+
+# debugger support
+PA13 JTMS-SWDIO SWD
+PA14 JTCK-SWCLK SWD
+
+# spi bus, unused
+PB3 SPI1_SCK SPI1
+PB4 SPI1_MISO SPI1
+PB5 SPI1_MOSI SPI1
+
+# single rm3100 mag on I2C
+COMPASS RM3100 I2C:0:0x20 false ROTATION_NONE
+COMPASS RM3100 I2C:0:0x21 false ROTATION_NONE
+COMPASS RM3100 I2C:0:0x22 false ROTATION_NONE
+COMPASS RM3100 I2C:0:0x23 false ROTATION_NONE
```

Comment:
The address isn't fixed?

Reply:
need to probe all 4 for different hw revs